helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -269,7 +269,10 @@ public ZKHelixManager(String clusterName, String instanceName, InstanceType inst
     }
 
     _stateListener = stateListener;
+    // read cloud config from ZK and set cloudConfig in HelixManagerProperty
     _helixManagerProperty = helixManagerProperty;
+    _helixManagerProperty
+        .setHelixCloudProperty(HelixPropertyFactory.getCloudConfig(_zkAddress, _clusterName));
```

Comment:
Maybe not related, what the difference between this and using `_configAccessor.getCloudConfig()`?

Reply:
`_configAccessor.getCloudConfig()` also returns cloud config. However, _configAccessor is initiated in `connect()` not the zkHelixManager constructor.  So in constructor we need to use `HelixPropertyFactory.getCloudConfig`.